### ISSUE TITLE
Issue in order details products loop

### ISSUE THREAD

**Issue by Kandax86** [Oksydan/falcon#147]
Hi,
there is an issue in the theme in : 
templates/customer/_partials/product-table-line.tpl

When you delete a combination in BO, it deletes the lines in cart_product. Si you cant retrieve cover and attributes anymore because of the look in orderPresenter => orderLazyArray  line 203 👍 
```
foreach ($cartProducts['products'] as $cartProduct) {
                if (($cartProduct['id_product'] === $orderProduct['id_product'])
                    && ($cartProduct['id_product_attribute'] === $orderProduct['id_product_attribute'])
                ) {
                    if (isset($cartProduct['attributes'])) {
                        $orderProduct['attributes'] = $cartProduct['attributes'];
                    } else {
                        $orderProduct['attributes'] = [];
                    }
                    $orderProduct['cover'] = $cartProduct['cover'];
                    $orderProduct['default_image'] = $cartProduct['default_image'];
                    $orderProduct['unit_price_full'] = $cartProduct['unit_price_full'];
                    break;
                }
            }
```
you need to add conditions to prevent fatal errors on the tpl because you can't have references to find cover from cart products :
```
<td class="product-line__cell product-line__cell--prod">
  <div class="d-flex">
  {if !empty($product.cover)}
    <img
      {generateImagesSources image=$product.cover size='cart_default' lazyload=false}
      alt="{$product.name|escape:'quotes'}"
      class="product-line__img rounded img-fluid mr-3"
      width="{$product.cover.bySize.cart_default.width}"
      height="{$product.cover.bySize.cart_default.height}">
    {else}
      <img src="{$urls.no_picture_image.bySize.cart_default.url}" loading="lazy" width="{$urls.no_picture_image.bySize.cart_default.width}" height="{$urls.no_picture_image.bySize.cart_default.height}"
        class="order-products__image card-img-top w-auto" />
    {/if}
    
      {assign var="productAttrs" value=[]}
      {if !empty($product.attributes)}
      {foreach from=$product.attributes key="attribute" item="value"}
        {$productAttrs[]= "- `$attribute` : `$value`"}
      {/foreach}

      {$productAttrs = ""|implode:$productAttrs}
      {/if}
      <div class="product-line__title mb-2 my-auto">
        <div class="h6 mb-0">
          {if !empty($product.attributes) && $product.attributes|count > 0}
            {$product.name|replace:$productAttrs : ''}
          {else}
            {$product.name}
          {/if}
        </div>
        {if !empty($product.reference)}
          <div class="color-grey-600">
            {l
              s='ref : %ref%'
              d='Shop.Theme.Customeraccount'
              sprintf=['%ref%' =>$product.reference]
            }
          </div>
        {/if}
      </div>
  </div>

  {*if $product.attributes}
    <ul class="product-line__attributes mb-0">
      {foreach from=$product.attributes key="attribute" item="value"}
        <li class="product-line__attribute text-muted small">
          <span>{$attribute}</span>: <span class="font-weight-bold">{$value}</span>
        </li>
      {/foreach}
    </ul>
  {/if*}

</td>

<td class="product-line__cell product-line__cell--price" data-title="{l s='Price' d='Shop.Theme.Catalog'}">
  <div class="product-line__price">
    {$product.price}
  </div>
</td>

<td class="product-line__cell product-line__cell--qty" data-title="{l s='Quantity' d='Shop.Theme.Catalog'}">
  <span class="product-line__price">{$product.quantity}</span>
</td>

{if $page.page_name == 'order-detail' && $order.details.is_returnable && isset($product.qty_returned)}
  <td class="product-line__cell product-line__cell--returned" data-title="{l s='Returned' d='Shop.Theme.Customeraccount'}">
    <div class="product-line__price">
      {$product.qty_returned}
    </div>
  </td>
{/if}

<td class="product-line__cell product-line__cell--total" data-title="{l s='Total' d='Shop.Theme.Checkout'}">
  <div class="product-line__price">
    {if isset($product.is_gift) && $product.is_gift}
      {l s='Gift' d='Shop.Theme.Checkout'}
    {else}
      {$product.total}
    {/if}
  </div>
</td>
```

regards,

**Comment by Oksydan** [comment]
Hi @Kandax86,

thank you for creating an issue.
You are right and actually there should be used `default_image` instead of `cover`.

I will add this in `3.0.1` since version `3.0.0` is around the corner.

**Comment by Oksydan** [comment]
Hi @Kandax86,

problem has been fixed [here](https://github.com/Oksydan/falcon/commit/2ca3674096b9c0b69d2421e80cbac9fc549b34e1)
It will be available in version `3.0.1`.

**Comment by Oksydan** [comment]
Hi @Kandax86,

new minor version has been released https://github.com/Oksydan/falcon/releases/tag/v3.0.1
I am closing this issue.
